### ISSUE TITLE
Underlying autograd mechanism

### ISSUE THREAD

**Issue by ctongfei** [ctongfei/nexus#16]
C. Elliott (2018): The simple essence of automatic differentiation. arXiv. ?

**Comment by doofin** [comment]
Do you use the reverse mode diff techniques in this paper? BTW,forward AD is easy to understand with dual numbers ,would you like to recommend some papers on reverse mode ad?

**Comment by ctongfei** [comment]
No. Right now `nexus-diff` uses Wengert tape-based mechanism (the same as PyTorch). The paper above is a candidate for a future implementation.

http://www.cs.cmu.edu/~wcohen/10-605/notes/autodiff.pdf -- this note on reverse-mode autodiff is pretty good
